pkg/router/routeapihelpers/validation.go

```diff
@@ -251,6 +252,14 @@ func ExtendedValidateRoute(route *routev1.Route) field.ErrorList {
 			if len(keyBytes) == 0 {
 				result = append(result, field.Invalid(tlsFieldPath.Child("key"), "", "no key specified"))
 			} else {
+				// Validate that final key contains only private key, and cert contains only public keys
```

Comment:
Would it be worth it here to also call [cert.parseCertsPEM ](https://github.com/kubernetes/client-go/blob/master/util/cert/pem.go#L35) like we do for the other certs in the `tlsConfig`?

Reply:
Actually, never mind,  we already call it on Line 226.

Reply:
I don't think the comment is describing what's happening.  The certBytes is being checked that it only contains CERTIFICATE blocks.  Doesn't [splitCertKey](https://github.com/openshift/router/blob/master/pkg/router/routeapihelpers/validation.go#L146-L169) do what you want to do here?

Reply:
not really, because on the "else" condition the keyBytes can be empty, and then you append the tlsConfig.Key `keyBytes = append(keyBytes, []byte(tlsConfig.Key)...)` which you didn't really validated yet.

The certBytes indeed after the split may have just CERTIFICATE blocks, but I still wanted to guarantee that we were not leaking any "bug" from the split into the PEM content, that's why I have executed it again

Reply:
(I mean, this is more like an additional defensive check)

Reply:
I still think you need to check if keyBytes and/or certBytes is nil before calling validatePEMContent.  If either is nil, there is no error result appended that would report that.

Reply:
added a check for len < 0 on validatePEMContent